packages/synapse-react-client/src/components/TableColumnSchemaEditor/Validators/EntityIdSchema.ts

```diff
@@ -0,0 +1,9 @@
+import { z } from 'zod'
+
+/**
+ * Validates and converts a value to a valid Synapse ENTITYID value
+ */
+export const entityIdSchema = z.union([
+  z.string().regex(/^(syn)?\d+$/),
```

Comment:
Even though this regex is straightforward, you should probably add a comment that explains in english (most regex test services provide this copy if you don't want to write it yourself)